docs/add-runtime.md

```diff
@@ -0,0 +1,254 @@
+### 2.4 The Error Schema
+All errors that occur during the execution of a user's function **MUST** be returned using this JSON Object otherwise OpenRuntimes will be unable to parse them for the user.
+```json
+{
+    "code": 500, // (Int) Use 404 if function not found or use 401 if the x-internal-challenge check failed.
+    "message": "Error: Tried to divide by 0 \n /usr/code/index.js:80:7", // (String) Try to return a stacktrace and detailed error message if possible. This is shown to the user.
+}
+```
+
+### 2.5 Writing your Dockerfile
+The Dockerfile is very important as it's the environment you are creating to run build the runtime and also run the code if you are writing an Interpreted Runtime (Compiled runtimes will use an alpine or ubuntu image to run the compiled executable.)
+
+The first thing you need to do is find a docker image to base your runtime off, You can find these at [Docker Hub](https://hub.docker.com). If possible try to use verified official builds of the language you are creating a runtime for.
+
+Next in your Dockerfile at the start add the docker image you want to base it off at the top like so:
+```bash
+FROM Dart:2.12 # Dart is used as an example.
+```
+This will download and require the image when you build your runtime and allow you to use the toolset of the language you are building a runtime for.
+
+Create a user and group for the runtime, this user will be used to both build and run the code:
+```bash
+RUN groupadd -g 2000 openruntimes \
+&& useradd -m -u 2001 -g openruntimes openruntimes
+```
+
+then create the folders you will use in your build step:
+```bash
+RUN mkdir -p /usr/local/src/
+RUN mkdir -p /usr/code
+RUN mkdir -p /usr/workspace
+RUN mkdir -p /usr/builtCode
+```
+
+Next copy your source code and set the working directory for the image like so:
+```
+WORKDIR /usr/local/src
+COPY . /usr/local/src
+```
+
+Next, you want to make sure you are adding execute permissions to any scripts you may run, the main ones are `build.sh` and `launch.sh`. You can run commands in Dockerfile's using the `RUN` prefix like so:
+```
+RUN chmod +x ./build.sh
+RUN chmod +x ./launch.sh
+```
+Note: Do not chmod a `launch.sh` file if you don't have one.
+
+If needed use the `RUN` commands to install any dependencies you require for the build stage.
+
+Next set the permissions for the user you created so your build and run step will have access to them:
+```
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/local/src"]
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/code"]
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/workspace"]
+RUN ["chown", "-R", "openruntimes:openruntimes", "/usr/builtCode"]
+```
+
+Finally, you'll add a `CMD` command. For an interpreted language this should be:
+```
+CMD ["/usr/local/src/launch.sh"]
+```
+Since this will use your launch script when the runtime starts.
+
+For a compiled language this must be:
+```
+CMD ["tail", "-f", "/dev/null"]
+```
+so the build steps can be run.
```

Comment:
You can remove this, since we're using the `start.sh` script as the entrypoint for all the runtimes